common/tablefunctions.lua

```diff
@@ -256,3 +256,28 @@ if not pairsByKeys then
 		return iter
 	end
 end
+
+if not table.countMem then 
+	--- Estimate how much memory a table (recursively) consumes
+	---@generic V
+	---@param tbl V[]|table<any, V>
+	---@param maxdepth
+	---@return integer number of bytes estimated by this table.
+	function table.countMem(tbl, maxdepth)
+		if maxdepth == 0 then return 0 end
+		maxdepth = maxdepth or 0
+		local ttype = type(tbl)
+		local myMem = 0
```

Comment:
shouldn't this be at least 16 for the overhead of keeping pointers to the hash and array parts? perhaps more

Reply:
At that point you are past maxdepth, so it doesnt count. Pointers are calced on next pairs call

Reply:
is depth related here? even very shallow tables have overhead.
* `{ x = 123 }`, in addition to storing the single key and single value, has to store a pointer to the hashmap that keeps those keys/values
* `{}` has zero depth so surely doesn't reach maxdepth, but has a non-zero actual size (because you have null pointers to hashmap/array in its internals, and also the table itself is represented by some hash or pointer in whatever table it is itself contained by)

Reply:
Maxdepth is pretty much so you can bail out of a recursion problem. I dont expect tables to be be nested more than 4-5. anyway. Ill revamp this a little bit, because this is mainly for GC pressure. And GC pressure is more related to number of tables than to the number of elements in them.